File: Msys_Programs/MSys_Projects/LeetCode/GoLang/Binary_Search.go
package main

import "fmt"

func search(nums []int, target int) int {
	start := 0
	end := len(nums) - 1

	for start <= end {
		mid := (end + start) >> 1
		if nums[mid] == target {
			return mid
		} else if nums[mid] < target {
			start = mid + 1
		} else {
			end = mid - 1
		}
	}
	return -1
}

func main() {
	O := []int{2, 3, 5, 6, 7, 9}
	fmt.Println(search(O, 5))
	fmt.Println(search(O, 1))
	fmt.Println(search(O, 8))

}

File: Msys_Programs/MSys_Projects/LeetCode/Python/Binary_Search.py
class Solution:
    def search(self, nums, target):
        index = 0
        while index < len(nums):
            if nums[index] == target:
                return index
            index = index + 1
        return -1

O = Solution()
print(O.search([2,4,5,6,7], 5))
print(O.search([2,4,5,6,7], 8))
print(O.search([2,4,5,6,7], 6))